crates/indexer/src/geyser/accounts/receipt.rs

```diff
@@ -120,15 +121,23 @@ pub(crate) async fn process_bid_receipt(
     client
         .db()
         .run(move |db| {
+            let bid_receipt_exists = select(exists(
+                bid_receipts::table.filter(bid_receipts::address.eq(row.address.clone())),
+            ))
+            .get_result::<bool>(db);
             insert_into(bid_receipts::table)
                 .values(&row)
                 .on_conflict(bid_receipts::address)
                 .do_update()
                 .set(&row)
-                .execute(db)
+                .execute(db)?;
+            if Ok(true) == bid_receipt_exists || row.purchase_receipt.is_some() {
```

Comment:
This if statement does nothing, `return` in Rust closures is more like `next` in Ruby closures.

Reply:
whats the suggestion here @ray-kast ?